components/Orders/ProposalOpenOrderRow.tsx

```diff
@@ -46,33 +48,17 @@ export function ProposalOpenOrderRow({ order }: { order: OpenOrdersAccountWithKe
   const [isSettling, setIsSettling] = useState<boolean>(false);
 
   const handleCancel = useCallback(async () => {
-    if (!proposal || !markets) return;
+    if (!markets || !proposal) return;
 
     const marketAccount = proposal.account.openbookPassMarket.equals(order.account.market)
       ? { publicKey: proposal.account.openbookPassMarket, account: markets.pass }
       : { publicKey: proposal.account.openbookFailMarket, account: markets.fail };
-    const txs = await cancelOrderTransactions(
-      new BN(order.account.accountNum),
-      marketAccount,
-    );
-
-    if (!wallet.publicKey || !txs) return;
 
     try {
       setIsCanceling(true);
-
-      //settle funds
-      const pass = order.account.market.equals(proposal.account.openbookPassMarket);
-      const settleTxs = await settleFundsTransactions(
-        order.account.accountNum,
-        pass,
-        proposal,
-        marketAccount,
-      );
-
-      await sender.send([...txs, ...settleTxs]);
-      // We already return above if the wallet doesn't have a public key
-      await fetchOpenOrders(wallet.publicKey!);
+      await cancelOrder(order);
+      await fetchBalance(marketAccount.account.baseMint);
```

Comment:
Batching this into a get multiple saves on raw # of requests, and likely we can use what we've setup below given the fact that we have a function that should update all balances based on account update? Not saying it needs to happen now, just pointing out where we're not being careful with what we're ultimately trying to accomplish.

Reply:
Oh yeah good point, we could just update the balance in state rather than refetch. Let me look at that.